crates/wdk-build/src/resource_compile.rs

```diff
@@ -0,0 +1,217 @@
+use std::{
+    env,
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use cargo_metadata::MetadataCommand;
```

Comment:
I think this file should probably be restructured such that it is some ResourceCompile builder, rather than a set of free functions. I think the process of compiling the rc file should be separated from the process of writing the rc file from a template to filesystem. This way, we could reuse it for different rc files, if needed